orchestration/src/main/java/com/sap/ai/sdk/orchestration/DpiMasking.java

```diff
@@ -74,17 +77,40 @@ public DpiMasking withEntities(
       val entitiesList = new ArrayList<DPIEntities>();
       entitiesList.add(entity);
       entitiesList.addAll(Arrays.asList(entities));
-      return new DpiMasking(maskingMethod, entitiesList);
+      return new DpiMasking(maskingMethod, entitiesList, false, List.of());
     }
   }
 
+  /**
+   * If grounding is used, the input text will be masked.
+   *
+   * @return A new {@link DpiMasking} instance
+   */
+  @Nonnull
+  public DpiMasking withMaskGroundingEnabled() {
+    return new DpiMasking(maskingMethod, entities, true, allowList);
+  }
+
+  /**
+   * Set words that should not be masked.
+   *
+   * @param allowList List of strings that should not be masked
+   * @return A new {@link DpiMasking} instance
+   */
+  @Nonnull
+  public DpiMasking withAllowList(@Nonnull final List<String> allowList) {
+    return new DpiMasking(maskingMethod, entities, maskGroundingInput, allowList);
+  }
+
   @Nonnull
   @Override
   public MaskingProviderConfig createConfig() {
     val entitiesDTO = entities.stream().map(it -> DPIEntityConfig.create().type(it)).toList();
     return DPIConfig.create()
         .type(SAP_DATA_PRIVACY_INTEGRATION)
         .method(maskingMethod)
-        .entities(entitiesDTO);
+        .entities(entitiesDTO)
+        .maskGroundingInput(DPIConfigMaskGroundingInput.create().enabled(maskGroundingInput))
```

Comment:
(Question)

are the following two lines semantically equal?
```diff
- .maskGroundingInput(null) // before
+ .maskGroundingInput(DPIConfigMaskGroundingInput.create().enabled(false)) // after
```

Reply:
Yes